packages/nextjs/src/app-router/server/ClerkProvider.tsx

```diff
@@ -8,14 +10,43 @@ import { ClientClerkProvider } from '../client/ClerkProvider';
 import { initialState } from './auth';
 import { getScriptNonceFromHeader } from './utils';
 
-export function ClerkProvider(props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange'>) {
+export async function ClerkProvider(
+  props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange'>,
+) {
   const { children, ...rest } = props;
   const state = initialState()?.__clerk_ssr_state as InitialState;
   const cspHeader = headers().get('Content-Security-Policy');
 
+  const providerProps = { ...mergeNextClerkPropsWithEnv(rest) };
+
+  if (!providerProps.publishableKey) {
+    const ephemeralAccount = await fetchEphemeralAccount();
+    const cookieExpiresAt = cookies().get(constants.QueryParameters.EphemeralExpiresAt)?.value;
+    const cookiePublishableKey = cookies().get(constants.QueryParameters.EphemeralPublishableKey)?.value;
+    const cookieSecretKey = cookies().get(constants.QueryParameters.EphemeralSecretKey)?.value;
+
+    const stale =
+      cookieExpiresAt !== String(ephemeralAccount.expiresAt) ||
+      cookiePublishableKey !== ephemeralAccount.publishableKey ||
+      cookieSecretKey !== ephemeralAccount.secretKey;
+
+    if (stale) {
+      const params = new URLSearchParams({
+        [constants.QueryParameters.EphemeralExpiresAt]: String(ephemeralAccount.expiresAt),
+        [constants.QueryParameters.EphemeralPublishableKey]: ephemeralAccount.publishableKey,
+        [constants.QueryParameters.EphemeralSecretKey]: ephemeralAccount.secretKey,
+      });
+
+      redirect(`?${params}`);
+    }
```

Comment:
Yeah, think this was some edge case handling @ayrton had added to prevent the front end from being in a different state to the backend.

I think we can probably remove this in favor of the state that is propagated from the middleware.